packages/modelserver-client/src/utils/patch-utils.ts

```diff
@@ -0,0 +1,209 @@
+            ...attributes
+        }
+    };
+}
+
+/**
+ * Create an AddOperation, to add an existing object of the specified type, in the specified parent.
+ * @param modeluri the uri of the model to edit
+ * @param parent the parent in which the element should be added
+ * @param feature the property of the parent in which the element should be added
+ * @param value the element to add
+ * @returns The Json Patch AddOperation to add the element in the parent.
+ */
+export function add(modeluri: string, parent: ModelServerObjectV2, feature: string,
+    value: ModelServerObjectV2 | ModelServerReferenceDescriptionV2): AddOperation<ModelServerObjectV2> {
+    return {
+        op: 'add',
+        path: getPropertyPath(modeluri, parent, feature),
+        value: {
+            $type: value.$type,
+            $id: getObjectPath(modeluri, value)
+        }
+    };
+}
+
+/**
+ * Create a RemoveOperation, to delete an object from the model.
+ * @param modeluri the uri of the model to edit
+ * @param object the object to remove from the model
+ * @returns The Json Patch RemoveOperation to remove the element.
+ */
+export function deleteElement(modeluri: string, object: ModelServerObjectV2): RemoveOperation {
+    return {
+        op: 'remove',
+        path: getObjectPath(modeluri, object)
+    };
+}
+
+/**
+ * Create a RemoveOperation, to remove a value from a list.
+ * @param modeluri the uri of the model to edit
+ * @param object the object from which a value will be removed
+ * @param feature the property from which a value will be removed
+ * @param index the index of the value to remove
+ */
+export function removeValueAt(modeluri: string, object: ModelServerObjectV2, feature: string, index: number): RemoveOperation {
+    return {
+        op: 'remove',
+        path: getPropertyPath(modeluri, object, feature, index)
+    };
+}
+
+/**
+ * Create a RemoveOperation, to remove a value from a list.
+ * @param modeluri the uri of the model to edit
+ * @param object the object from which a value will be removed
+ * @param feature the property from which a value will be removed
+ * @param value the value to remove
+ */
+export function removeValue(modeluri: string, object: ModelServerObjectV2, feature: string, value: AnyObject): RemoveOperation | undefined {
+    const index = findIndex(object, feature, value);
+    if (index >= 0) {
+        return removeValueAt(modeluri, object, feature, index);
+    }
+    return undefined;
+}
+
+/**
+ * Create a RemoveOperation, to delete an object from the model.
+ * @param modeluri the uri of the model to edit
+ * @param objectToRemove the object to delete from the model
+ */
+export function removeObject(modeluri: string, objectToRemove: ModelServerObjectV2): RemoveOperation {
+    return {
+        op: 'remove',
+        path: getObjectPath(modeluri, objectToRemove)
+    };
+}
+
+export function getObjectPath(modeluri: string, object: ModelServerObjectV2 | ModelServerReferenceDescriptionV2): string {
```

Comment:
Hmm. Why is this even exported? It's an implementation detail of the higher-level functions defined above. I think it should be local only (But it can still be documented :) especially since we now support both EMF-like paths and standard Json Pointer paths)

Now I need to remember why I exported them in the first place... I'm pretty sure they weren't meant to be.

Reply:
Removing export AND provide documentation, what a treat, thank you ;-)